pkg/apiserver/rest/dualwriter.go

```diff
@@ -212,6 +212,9 @@ var defaultConverter = runtime.UnstructuredConverter(runtime.DefaultUnstructured
 
 // Compare asserts on the equality of objects returned from both stores	(object storage and legacy storage)
 func Compare(storageObj, legacyObj runtime.Object) bool {
+	if storageObj == nil || legacyObj == nil {
```

Comment:
hm, should `Compare(nil, nil)` be true?

Reply:
it should be true if both are nil, but it should be false if either are, no?

Reply:
Yup, so you need to return false if only one or the other is nil, but true if they are both nil
```suggestion
	if storageObj == nil || legacyObj == nil {
		return storageObj == nil && legacyObj == nil
```

Reply:
correct, I was seeing an `AND` there, for some reason 🤦‍♀️